compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -900,8 +903,27 @@ class Semantic {
             // It's always safe to approximate them with `Cold`.
             Result(Cold, Nil)
         else
-          default()
-
+          sym.defTree match {
+            case vdef: ValDef => {
+              // resolve this for local variable
+              val enclosingClass = sym.owner.enclosingClass.asClass
+              val thisValue2 = resolveThis(enclosingClass, thisV, klass, source)
+              thisValue2 match {
+                case Hot => Result(Hot, Errors.empty)
+                case Cold => {
+                  val error = AccessCold(sym, source, trace.toVector)
+                  Result(Hot, error :: Nil)
+                }
+                case addr: Addr => {
+                  val res = eval(vdef.rhs, addr, klass)
+                  if res.value.promote("Try promote", source).isEmpty then Result(Hot, Errors.empty) else res
+                }
+                case _ => ???
```

Comment:
How do we deal with other possible cases for thisV, like Fun or RefSet? Are they possible in this case?